sled-agent/src/sim/storage.rs

```diff
@@ -258,28 +299,34 @@ impl CrucibleDataInner {
             bail!("failure creating running snapshot");
         }
 
+        if self.get_snapshot_for_region(id, name).is_none() {
+            bail!("cannot create running snapshot, snapshot does not exist!");
+        }
+
         let id = Uuid::from_str(&id.0).unwrap();
+        let port_number = self.get_free_port();
 
         let map =
             self.running_snapshots.entry(id).or_insert_with(|| HashMap::new());
 
         // If a running snapshot exists already, return it - this endpoint must
         // be idempotent.
         if let Some(running_snapshot) = map.get(&name.to_string()) {
+            self.used_ports.remove(&port_number);
```

Comment:
Nit: Could you put the `let port_number = ..` below this if and then you don't have to remove it here.

Reply:
Good catch! Done in 4b705aff6